great_expectations_cloud/agent/actions/agent_action.py

```diff
@@ -1,22 +1,14 @@
 from __future__ import annotations
 
 from abc import abstractmethod
-from typing import TYPE_CHECKING, Generic, Sequence, TypeVar
+from typing import TYPE_CHECKING, Generic, TypeVar
 
-from pydantic.v1 import BaseModel
-
-from great_expectations_cloud.agent.models import CreatedResource, Event
+from great_expectations_cloud.agent.models import ActionResult, Event
 
 if TYPE_CHECKING:
     from great_expectations.data_context import CloudDataContext
 
 
-class ActionResult(BaseModel):
-    id: str
-    type: str
-    created_resources: Sequence[CreatedResource]
-
-
```

Comment:
A lot of the diffs in this file are due to this change--I moved the `ActionResult` model to `models.py`. Reason: I wanted to make the added `job_duration_milliseconds` property optional, but there is a ruff rule that requires us to use `float | None` syntax for this but then python 3.8/9 doesn't support that, so tests were failing.

Moving `ActionResult` to `models.py` means that this issue will be temporarily ignored until we move off of python 3.8/9 and then it'll be easy to address.

The alternative would have been to ignore the ruff issue, but that might have been harder to notice when the time comes to upgrade. The drawback is 1) the large diff here and 2) a larger generic models file.